lib/sudo-cli/src/lib.rs

```diff
@@ -244,220 +75,268 @@ pub struct SudoOptions {
-            // If the last external arg is the same as this arg, we know that the separator must go
-            // before this arg.
-            if let Some(last_external) = external_args.next() {
-                if last_arg == last_external {
-                    continue;
+    /// parse command line arguments from the environment and handle errors
+    pub fn parse() -> SudoOptions {
+        match Self::try_parse_from(std::env::args()) {
+            Ok(options) => {
+                if options.help {
+                    eprintln!("{HELP_MSG}");
+                    std::process::exit(1);
                 }
-            }
-            // Otherwise, this arg is already a non-external arg.
-            index = Some(i + 1);
-            break;
-        }
 
-        if let Some(mut index) = index {
-            // There is one last catch. In this command `sudo FOO=1 env BAR=2 ls`, `FOO=1` is not
-            // part of the external command. So we need to ignore the first environment variable
-            // declarations before putting the separator.
-            while vec_env_vars[index] {
-                index += 1;
+                options
+            }
+            Err(e) => {
+                eprintln!("{e}\n{USAGE_MSG}");
+                std::process::exit(1);
             }
-            // Then we insert the separator and parse the arguments again. This will not recurse
-            // indefinitely because we don't do any of this extra logic if there was a separator.
-            vec_args.insert(index, "--".to_owned());
-            Self::try_parse_from(vec_args)
-        } else {
-            // If `index` were `None`, it would mean that all the args were environment variable
-            // declarations or part of the external args. But this cannot happen because the first
-            // argument should be the name of the binary being run and that is neither an
-            // environment variable declaration nor a external argument
-            unreachable!()
         }
     }
 
-    pub fn parse() -> Self {
-        match Self::try_parse_from(std::env::args()) {
-            Ok(options) => options,
-            Err(err) => {
-                eprintln!("{err}");
-                exit(1);
+    /// parse an iterator over command line arguments
+    pub fn try_parse_from<I, T>(iter: I) -> Result<Self, &'static str>
+    where
+        I: IntoIterator<Item = T>,
+        T: Into<String> + Clone,
+    {
+        let mut options: SudoOptions = SudoOptions::default();
+        let arg_iter = Self::normalize_arguments(iter.into_iter().map(Into::into))?
+            .into_iter()
+            .peekable();
+
+        for arg in arg_iter {
+            match arg {
+                SudoArg::Flag(flag) => match flag.as_str() {
+                    "-A" | "--askpass" => {
+                        options.askpass = true;
+                    }
+                    "-b" | "--background" => {
+                        options.background = true;
```

Comment:
I'm not sure this is the right time, but seeing we don't (yet) support these kinds of arguments, do we want to provide any user feedback at this point indicating that we don't?